File: python/codeeval/028_string_search.py
import sys
import re

def ss(test):
    i, s = test.strip().split(',')
    se = re.sub(r'\*', '.*', s)
    se = re.sub(r'(\\\.\*)', '\*', se)
    m = re.search(se, i)
    if m:
        print('true')
    else:
        print('false')

with open(sys.argv[1], 'r') as test_cases:
    for test in test_cases:
        ss(test)

File: ruby/codeeval/028_string_search.rb
def ss(line)
    t = line.strip().split(',')
    i = t[0]
    s = t[1]
    se = s.gsub('*', '.*')
    se = se.gsub('\.*', '\*')
    m = i =~ Regexp.new(se)
    return !!m
end

File.open(ARGV[0]).each_line do |line|
    puts(ss(line))
end
